pkg/ccl/importccl/exportparquet.go

```diff
@@ -181,57 +210,356 @@ func newParquetColumn(typ *types.T, name string, nullable bool) (parquetColumn,
+	schemaEl.RepetitionType = parquet.FieldRepetitionTypePtr(parquet.FieldRepetitionType_OPTIONAL)
 	if !nullable {
-		col.definition.SchemaElement.RepetitionType = parquet.FieldRepetitionTypePtr(parquet.FieldRepetitionType_REQUIRED)
+		schemaEl.RepetitionType = parquet.FieldRepetitionTypePtr(parquet.FieldRepetitionType_REQUIRED)
 	}
-	col.definition.SchemaElement.Name = col.name
+	schemaEl.Name = col.name
 
 	// MB figured out the low level properties of the encoding by running the goland debugger on
 	// the following vendor example:
 	// https://github.com/fraugster/parquet-go/blob/master/examples/write-low-level/main.go
 	switch typ.Family() {
 	case types.BoolFamily:
-		col.definition.SchemaElement.Type = parquet.TypePtr(parquet.Type_BOOLEAN)
+		schemaEl.Type = parquet.TypePtr(parquet.Type_BOOLEAN)
 		col.encodeFn = func(d tree.Datum) (interface{}, error) {
 			return bool(*d.(*tree.DBool)), nil
 		}
+		col.DecodeFn = func(x interface{}) (tree.Datum, error) {
+			return tree.MakeDBool(tree.DBool(x.(bool))), nil
+		}
 
 	case types.StringFamily:
-		col.definition.SchemaElement.Type = parquet.TypePtr(parquet.Type_BYTE_ARRAY)
-		col.definition.SchemaElement.LogicalType = parquet.NewLogicalType()
-		col.definition.SchemaElement.LogicalType.STRING = parquet.NewStringType()
-		col.definition.SchemaElement.ConvertedType = parquet.ConvertedTypePtr(parquet.ConvertedType_UTF8)
+		populateLogicalStringCol(schemaEl)
 		col.encodeFn = func(d tree.Datum) (interface{}, error) {
 			return []byte(*d.(*tree.DString)), nil
 		}
-
+		col.DecodeFn = func(x interface{}) (tree.Datum, error) {
+			return tree.NewDString(string(x.([]byte))), nil
+		}
+	case types.CollatedStringFamily:
+		populateLogicalStringCol(schemaEl)
+		col.encodeFn = func(d tree.Datum) (interface{}, error) {
+			return []byte(d.(*tree.DCollatedString).Contents), nil
+		}
+		col.DecodeFn = func(x interface{}) (tree.Datum, error) {
+			return tree.NewDCollatedString(string(x.([]byte)), typ.Locale(), &tree.CollationEnvironment{})
+		}
+	case types.INetFamily:
+		populateLogicalStringCol(schemaEl)
+		col.encodeFn = func(d tree.Datum) (interface{}, error) {
+			return []byte(d.(*tree.DIPAddr).IPAddr.String()), nil
+		}
+		col.DecodeFn = func(x interface{}) (tree.Datum, error) {
+			return tree.ParseDIPAddrFromINetString(string(x.([]byte)))
+		}
+	case types.JsonFamily:
+		schemaEl.Type = parquet.TypePtr(parquet.Type_BYTE_ARRAY)
+		schemaEl.LogicalType = parquet.NewLogicalType()
+		schemaEl.LogicalType.JSON = parquet.NewJsonType()
+		col.encodeFn = func(d tree.Datum) (interface{}, error) {
+			return []byte(d.(*tree.DJSON).JSON.String()), nil
+		}
+		col.DecodeFn = func(x interface{}) (tree.Datum, error) {
+			jsonStr := string(x.([]byte))
+			return tree.ParseDJSON(jsonStr)
+		}
+		// TODO (butler): make fine grain encoding of Int and Floats round trippable. For example,
+		// converting float32 to 64 in the float decoder isn't easily possible.
```

Comment:
The future versions of us who go to tackle this TODO may benefit from a bit more detail on the problems you ran into.